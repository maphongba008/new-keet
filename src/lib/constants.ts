import { GranularPermission } from 'expo-media-library'
import Config from 'react-native-config'
import { type Edge } from 'react-native-safe-area-context'

import { CALL_VIDEO_LIMIT_MEMBERS_GTE } from '@holepunchto/keet-store/store/call'
import { MemberCapabilities } from '@holepunchto/keet-store/store/member'

import { isIOS } from './platform'

export const APPLICATION_KEY = __DEV__ ? 'dev' : 'keet'
export const PEAR_PROTOCOL = Config.KEET_SCHEME_PEAR
export const ROOM_URL_PREFIX = `${PEAR_PROTOCOL}://`

export const CUSTOM_LINK_PROTOCOLS = ['holepunch://', 'punch://', PEAR_PROTOCOL]

// APP specific URLs
export const SUPPORT_URL = `${ROOM_URL_PREFIX}keet/yry7g7wa7bmrjd1xxr7ce64581ij3hozq7589xfgxjrm74kgp34o36pwp46ohw4u1w63o6is6sax6j3swgdi66dmti8p8nfuyypfwewbuy`
export const NEWS_URL = `${ROOM_URL_PREFIX}keet/yry5dqgiy967ffrrbi973ghfb4wsbtczezsomgp4z8s8nn68eyg5kgdyjc1b7dh4gbxkh9apmzn8szr3wdyosa8bdbmja47pp48pjmf7dh`
export const TWITTER_URL = 'https://twitter.com/keet_io'

export const KEET_URL = 'https://keet.io/'

export const CSAM_REPORT_URL = 'https://projectcport.com/report/'

// Push URL
export const ROOM_PUSH_URL_PREFIX = `${ROOM_URL_PREFIX}push/`
export const SHARE_CONTENT_URL = 'shareContentURL'
export const KEET_URL_PREFIX = `${Config.KEET_SCHEME_KEET}://`

export const INVITE_DURATION = {
  HOURS: '2h',
  DAYS: '2d',
  WEEKS: '2w',
  PERMANENT: '0',
}

// APP UI
export const SAFE_EDGES: readonly Edge[] = ['left', 'right', 'bottom']
export const BACK_DEBOUNCE_DELAY = 500
export const BACK_DEBOUNCE_OPTIONS = { leading: true, trailing: false }
export const BOTTOM_SHEET_ANIMATION_DURATION = 250
export const BOTTOM_SHEET_ANIMATION_REPLACE_TIMEOUT = 100

export const MAX_FONT_SIZE_MULTIPLIER = 1.5

export const CAPABILITIES = {
  CAN_WRITE: Number.parseInt(MemberCapabilities.CAN_WRITE, 10),
  CAN_INDEX: Number.parseInt(MemberCapabilities.CAN_INDEX, 10),
  CAN_MODERATE: Number.parseInt(MemberCapabilities.CAN_MODERATE, 10),
}

export const IGNORE_LOGS = [
  /^You seem to update pro/,
  /^Key "cancelled" in the/,
  /^Failed to get size/,
  '[Reanimated] Reading from `value` during component render. Please ensure that you do not access the `value` property or use `get` method of a shared value while React is rendering a component.',
]

// ROOM
export const LOBBY_PAGINATION_SCROLL_DEBOUNCE = 30
export const LOBBY_DECELERATION_RATE = isIOS ? 0.95 : 0.9
export const VIEWABLE_ITEM_DEBOUNCE_OPTIONS = { leading: false, trailing: true }
export const VIEWABLE_ITEM_DURATION = 200

// ROOM OPTION
export const ROOM_DEBUG_MENU_PASSWORD = 'thecakeisalie'
export const STATS_EXPANDED_MAX_HEIGHT = 300
export const STATS_MAX_TAPS = 10

// Profile
export const PROFILE_NAME_CHAR_LIMIT = 22

// CHAT
export const INPUT_DEBOUNCE_WAIT_TIME = 300
export const INPUT_DEBOUNCE_OPTIONS = { leading: false, trailing: true }

export const INVIEW_ITEMS_COUNT = 15

export const CHAT_BUBBLE_MAX_CHAR = 1500
export const QUOTE_NEST_LAYERS = 1

const PREVIEW_WIDTH = 50
const LINK_PREVIEW_WIDTH = 150

export const CHAT_INPUT_PREVIEW = {
  UNKNOWN_WIDTH: 150,
  PREVIEW_WIDTH,
  PREVIEW_HEIGHT: PREVIEW_WIDTH,
  LOADER_SIZE: PREVIEW_WIDTH / 2,
  LINK_PREVIEW_WIDTH,
  LINK_PREVIEW_HEIGHT: LINK_PREVIEW_WIDTH / 3,
}

// CALL
export const REMOTE_SYNC_DELAY = 1000

// MEDIA
export const GALLERY_WRITE_ONLY_MEDIA = false
export const GALLERY_GRANULAR_PERMISSION: GranularPermission[] = [
  'photo',
  'video',
]
export const MAX_SHARE_COUNT = 10

export const MOBILE_CALL_VIDEO_LIMIT_MEMBERS = CALL_VIDEO_LIMIT_MEMBERS_GTE - 1

export const WAVEFORM_HEIGHT = 30
export const WAVEFORM_MIN_HEIGHT = 2
export const WAVEFORM_SPACE = 0.2

export const DEFAULT_SVG_DIMENSIONS = {
  width: 800,
  height: 800,
}

export const ONE_SECOND_IN_MILLISECONDS = 1000
export const ONE_MINUTE_IN_MILLISECONDS = ONE_SECOND_IN_MILLISECONDS * 60
export const ONE_HOUR_IN_MILLISECONDS = ONE_MINUTE_IN_MILLISECONDS * 60

export const MOBILE_AUDIO_MAX_DURATION_TO_GET_WAVEFORM =
  ONE_MINUTE_IN_MILLISECONDS * 5 // 5 Minutes
export const MOBILE_AUDIO_MAX_BYTE_TO_GET_WAVEFORM = 10 ** 6 * 10 // 10 MB

export const NO_PERMISSION = 'NO_PERMISSION'
