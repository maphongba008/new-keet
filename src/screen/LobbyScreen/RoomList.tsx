// desktop equivalent https://github.com/holepunchto/keet-desktop/blob/main/src/components/room/room-list.js
import React, { memo, useCallback, useRef } from 'react'
import {
  Dimensions,
  KeyboardAvoidingView,
  StyleSheet,
  Text,
  View,
  VirtualizedList,
} from 'react-native'
import { useDispatch, useSelector } from 'react-redux'
import isEqual from 'react-fast-compare'
import { useFocusEffect, useScrollToTop } from '@react-navigation/native'
import _compact from 'lodash/compact'
import _isNaN from 'lodash/isNaN'
import _size from 'lodash/size'
import _throttle from 'lodash/throttle'

import {
  getRoomListActualIds,
  getRoomListInView,
  getRoomListIsCache,
  getRoomListIsError,
  getRoomListIsLoading,
  getRoomListSearchActive,
  getRoomListSearchIds,
  leaveRoomAsk,
  setItemsInView,
  switchRoomSubmit,
} from '@holepunchto/keet-store/store/room'

import {
  getIsStoreReady,
  getLobbyUpdating,
  getShareContent,
} from 'reducers/application'

import { showOptionsSheet } from 'component/AppBottomSheet/AppBottomSheet.Store'
import { KeetLoading } from 'component/Loading'
import { NavBar, NavBarLoading } from 'component/NavBar'
import { useTheme } from 'component/theme'
import { APPIUM_IDs } from 'lib/appium'
import { IS_SENTRY_ENABLED } from 'lib/build.constants'
import s from 'lib/commonStyles'
import {
  BACK_DEBOUNCE_DELAY,
  VIEWABLE_ITEM_DEBOUNCE_OPTIONS,
} from 'lib/constants'
import { useDeepEqualSelector } from 'lib/hooks/useDeepEqualSelector'
import { useLobbyResetRoomId, useNavigateToRoom } from 'lib/hooks/useRoom'
import { keyboardBehavior } from 'lib/keyboard'
import { localStorage } from 'lib/localStorage'
import { getState } from 'lib/store'

import { useStrings } from 'i18n/strings'

import RoomActionsBtn from './RoomActionsBtn'
import { RoomListEmpty } from './RoomList.Empty'
import { RoomListHeader } from './RoomList.Header'
import { RoomListItem } from './RoomList.Item'
import { RoomSearch } from './RoomSearch'

const screen = Dimensions.get('screen')
const roomListItemSize = 80
const roomListPageSize = Math.ceil(screen.height / roomListItemSize) - 1
const getItemCount = (data: any[]) => _size(data)
const getItem = (data: any[], index: number) => data[index]
const viewabilityConfig = {
  viewAreaCoveragePercentThreshold: 5,
  waitForInteraction: true,
}

export const RoomList = memo(() => {
  const actualRoomIds = useDeepEqualSelector(getRoomListActualIds)
  const isSearchActive = useSelector(getRoomListSearchActive)
  const searchRoomIds = useDeepEqualSelector(getRoomListSearchIds)
  const roomIds = isSearchActive ? searchRoomIds : actualRoomIds
  const shareContent = useDeepEqualSelector(getShareContent)
  const isCache = useSelector(getRoomListIsCache)
  const isStoreReady = useSelector(getIsStoreReady)
  const isLoading = useSelector(getRoomListIsLoading)
  const isError = useSelector(getRoomListIsError)
  const { bottom } = useDeepEqualSelector(getRoomListInView)
  const lobbyUpdating = useSelector(getLobbyUpdating)

  const dispatch = useDispatch()
  const strings = useStrings()
  const theme = useTheme()

  // scroll to the top when home tab-bar presses
  const scrollRef = useRef<VirtualizedList<string>>(null)
  useScrollToTop(scrollRef)
  useLobbyResetRoomId()

  // State will be reset when visit ID screen. It will have room data back, just not know the limits. So init again
  useFocusEffect(
    useCallback(() => {
      if (isLoading || !isStoreReady) return
      if (bottom === 0 || isSearchActive) {
        dispatch(setItemsInView({ top: 0, bottom: roomListPageSize }))
      }
    }, [bottom, dispatch, isLoading, isSearchActive, isStoreReady]),
  )

  const initiateRoomNavigation = useNavigateToRoom()
  const onPress = useCallback(
    (roomId: string) => {
      dispatch(switchRoomSubmit({ roomId }))
      initiateRoomNavigation(roomId)
    },
    [dispatch, initiateRoomNavigation],
  )

  const onLongPress = useCallback(
    (roomId: string, canLeaveRoom: boolean) => {
      const options = _compact([
        canLeaveRoom && {
          onPress: () => {
            dispatch(leaveRoomAsk(roomId))
          },
          title: strings.room.leave,
          icon: 'arrowRightFromBracket',
          iconColor: theme.color.danger,
          disabled: false,
          testID: APPIUM_IDs.lobby_bottomsheet_btn_leave,
        },
        IS_SENTRY_ENABLED && {
          onPress: () => {
            localStorage.setItem(`chat/last-seen-seq_${roomId}`, '0')
          },
          title: 'Reset unread message count',
          icon: 'trash',
          iconColor: theme.color.danger,
          disabled: false,
        },
      ])
      if (options.length) {
        showOptionsSheet({
          options,
        })
      }
    },
    [dispatch, strings.room.leave, theme.color.danger],
  )

  const renderRoomItem = useCallback(
    ({ item: roomId }: { item: string }) => {
      return (
        <RoomListItem
          roomId={roomId}
          onLongPress={onLongPress}
          onPress={onPress}
        />
      )
    },
    [onLongPress, onPress],
  )

  // eslint-disable-next-line react-hooks/exhaustive-deps
  const handleViewableItemsChanged = useCallback(
    _throttle(
      ({ viewableItems }: any) => {
        const { top, bottom: _bottom } = getRoomListInView(getState())
        const viewableTop = viewableItems[0]?.index
        const viewableBottom = viewableItems[_size(viewableItems) - 1]?.index
        if (
          _isNaN(viewableTop) ||
          _isNaN(viewableBottom) ||
          (viewableTop === top && viewableBottom === _bottom)
        )
          return
        dispatch(setItemsInView({ top: viewableTop, bottom: viewableBottom }))
      },
      BACK_DEBOUNCE_DELAY,
      VIEWABLE_ITEM_DEBOUNCE_OPTIONS,
    ),
    [dispatch],
  )

  const RoomListError = useCallback(() => {
    return (
      <View style={[s.container, s.centeredLayout]}>
        <Text style={theme.text.placeholder}>
          {strings.lobby.errorLoadingRooms} ðŸ˜•
        </Text>
      </View>
    )
  }, [strings.lobby.errorLoadingRooms, theme.text.placeholder])

  const keyExtractor = useCallback((roomId: string) => roomId, [])

  return (
    <KeyboardAvoidingView
      style={s.container}
      behavior={keyboardBehavior}
      keyboardVerticalOffset={0}
    >
      <NavBar
        showOfflineBar
        showTapToCallButton
        title=""
        left={null}
        middle={lobbyUpdating ? <NavBarLoading /> : <RoomSearch />}
        right={<RoomActionsBtn />}
        topTitle={shareContent.length > 0 ? strings.common.sendTo : null}
      />
      <View style={s.container}>
        {isLoading && !isCache ? (
          <View style={[s.container, s.centeredLayout]}>
            <KeetLoading />
          </View>
        ) : isError ? (
          <RoomListError />
        ) : (
          <>
            <VirtualizedList
              ref={scrollRef}
              initialNumToRender={roomListPageSize}
              decelerationRate={0.85}
              showsVerticalScrollIndicator={false}
              keyExtractor={keyExtractor}
              data={roomIds}
              ListHeaderComponent={RoomListHeader}
              renderItem={renderRoomItem}
              getItemCount={getItemCount}
              contentContainerStyle={styles.scrollViewContent}
              ListEmptyComponent={RoomListEmpty}
              getItem={getItem}
              windowSize={7}
              onViewableItemsChanged={handleViewableItemsChanged}
              viewabilityConfig={viewabilityConfig}
              keyboardDismissMode="on-drag"
              removeClippedSubviews={false}
            />
          </>
        )}
      </View>
    </KeyboardAvoidingView>
  )
}, isEqual)

const styles = StyleSheet.create({
  scrollViewContent: {
    flexGrow: 1,
  },
})
