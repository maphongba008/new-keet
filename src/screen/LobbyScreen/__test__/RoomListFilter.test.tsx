import { fireEvent, screen } from '@testing-library/react-native'
import { View } from 'react-native'

import { SHOW_ROOM_FILTER } from 'lib/build.constants'
import { renderWithProviders } from 'lib/testUtils'

import { RoomListFilter } from '../RoomList.Filter'

;(SHOW_ROOM_FILTER ? describe : describe.skip)('RoomListFilter', () => {
  test('render correctly', async () => {
    jest
      .mocked(View.prototype.measure)
      .mockImplementation((callback) => callback(0, 0, 0, 0, 0, 0))
    const element = <RoomListFilter />
    renderWithProviders(element, {
      preloadedState: {
        room: {
          list: {
            actualRoomIds: ['room1'],
            allRoomIds: ['room1'],
            roomTypesFilter: [0],
          },
        },
      },
    })
    expect(screen.toJSON()).toMatchSnapshot()
    // open filter
    fireEvent.press(screen.getByTestId('lobby_filter'))
    // click on group chats
    fireEvent.press(screen.getByTestId('Group Chats'))
    // check if selected value was changed
    expect(screen.getByTestId('lobby_selected_filter')).toHaveTextContent(
      'Group Chats',
    )
    expect(screen.toJSON()).toMatchSnapshot()
  })
})
