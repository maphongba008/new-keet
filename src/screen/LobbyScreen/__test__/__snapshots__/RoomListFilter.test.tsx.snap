// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RoomListFilter render correctly 1`] = `
<View
  style={
    [
      {
        "flex": 1,
      },
      {
        "paddingBottom": 4,
        "paddingHorizontal": 12,
        "paddingTop": 16,
      },
    ]
  }
>
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "flexDirection": "row",
        },
        {
          "justifyContent": "space-between",
        },
        {
          "alignItems": "center",
        },
      ]
    }
  >
    <Text
      accessibilityLabel="lobby_selected_filter"
      accessible={true}
      style={
        {
          "backgroundColor": "#0D0F16",
          "color": "#ffffff66",
          "fontFamily": "OpenSans-Bold",
          "fontSize": 16,
          "includeFontPadding": false,
        }
      }
      testID="lobby_selected_filter"
    >
      All rooms
    </Text>
    <View
      accessibilityLabel="lobby_filter"
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "borderRadius": 8,
          "opacity": 1,
          "padding": 8,
          "paddingVertical": 9,
        }
      }
      testID="lobby_filter"
    >
      <RNSVGSvgView
        align="xMidYMid"
        bbHeight={18}
        bbWidth={18}
        color="#ffffff"
        fill="#ffffff"
        focusable={false}
        height={18}
        meetOrSlice={0}
        minX={0}
        minY={0}
        style={
          [
            {
              "backgroundColor": "transparent",
              "borderWidth": 0,
            },
            {},
            {
              "flex": 0,
              "height": 18,
              "width": 18,
            },
          ]
        }
        testID="svg-icon-barsFilter"
        tintColor="#ffffff"
        vbHeight={24}
        vbWidth={24}
        width={18}
        xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M18.563 12a.56.56 0 0 1-.563.563H6a.563.563 0 0 1 0-1.126h12a.56.56 0 0 1 .563.563m3.187-5.062H2.25a.563.563 0 1 0 0 1.125h19.5a.562.562 0 1 0 0-1.126m-7.5 9h-4.5a.563.563 0 0 0 0 1.124h4.5a.562.562 0 1 0 0-1.125" fill="#fff"/></svg>
"
        xmlns="http://www.w3.org/2000/svg"
      >
        <RNSVGGroup
          fill={
            {
              "payload": 4294967295,
              "type": 0,
            }
          }
          propList={
            [
              "fill",
            ]
          }
        >
          <RNSVGPath
            d="M18.563 12a.56.56 0 0 1-.563.563H6a.563.563 0 0 1 0-1.126h12a.56.56 0 0 1 .563.563m3.187-5.062H2.25a.563.563 0 1 0 0 1.125h19.5a.562.562 0 1 0 0-1.126m-7.5 9h-4.5a.563.563 0 0 0 0 1.124h4.5a.562.562 0 1 0 0-1.125"
            fill={
              {
                "payload": 4294967295,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          />
        </RNSVGGroup>
      </RNSVGSvgView>
    </View>
  </View>
</View>
`;

exports[`RoomListFilter render correctly 2`] = `
[
  <View
    style={
      [
        {
          "flex": 1,
        },
        {
          "paddingBottom": 4,
          "paddingHorizontal": 12,
          "paddingTop": 16,
        },
      ]
    }
  >
    <View
      style={
        [
          {
            "flex": 1,
          },
          {
            "flexDirection": "row",
          },
          {
            "justifyContent": "space-between",
          },
          {
            "alignItems": "center",
          },
        ]
      }
    >
      <Text
        accessibilityLabel="lobby_selected_filter"
        accessible={true}
        style={
          {
            "backgroundColor": "#0D0F16",
            "color": "#ffffff66",
            "fontFamily": "OpenSans-Bold",
            "fontSize": 16,
            "includeFontPadding": false,
          }
        }
        testID="lobby_selected_filter"
      >
        Group Chats
      </Text>
      <View
        accessibilityLabel="lobby_filter"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "backgroundColor": "#222430",
            "borderRadius": 8,
            "opacity": 1,
            "padding": 8,
            "paddingVertical": 9,
          }
        }
        testID="lobby_filter"
      >
        <RNSVGSvgView
          align="xMidYMid"
          bbHeight={18}
          bbWidth={18}
          color="#26D2E8"
          fill="#26D2E8"
          focusable={false}
          height={18}
          meetOrSlice={0}
          minX={0}
          minY={0}
          style={
            [
              {
                "backgroundColor": "transparent",
                "borderWidth": 0,
              },
              {},
              {
                "flex": 0,
                "height": 18,
                "width": 18,
              },
            ]
          }
          testID="svg-icon-barsFilter"
          tintColor="#26D2E8"
          vbHeight={24}
          vbWidth={24}
          width={18}
          xml="<svg width="24" height="24" viewBox="0 0 24 24" fill="currentColor" xmlns="http://www.w3.org/2000/svg"><path d="M18.563 12a.56.56 0 0 1-.563.563H6a.563.563 0 0 1 0-1.126h12a.56.56 0 0 1 .563.563m3.187-5.062H2.25a.563.563 0 1 0 0 1.125h19.5a.562.562 0 1 0 0-1.126m-7.5 9h-4.5a.563.563 0 0 0 0 1.124h4.5a.562.562 0 1 0 0-1.125" fill="#fff"/></svg>
"
          xmlns="http://www.w3.org/2000/svg"
        >
          <RNSVGGroup
            fill={
              {
                "payload": 4280734440,
                "type": 0,
              }
            }
            propList={
              [
                "fill",
              ]
            }
          >
            <RNSVGPath
              d="M18.563 12a.56.56 0 0 1-.563.563H6a.563.563 0 0 1 0-1.126h12a.56.56 0 0 1 .563.563m3.187-5.062H2.25a.563.563 0 1 0 0 1.125h19.5a.562.562 0 1 0 0-1.126m-7.5 9h-4.5a.563.563 0 0 0 0 1.124h4.5a.562.562 0 1 0 0-1.125"
              fill={
                {
                  "payload": 4294967295,
                  "type": 0,
                }
              }
              propList={
                [
                  "fill",
                ]
              }
            />
          </RNSVGGroup>
        </RNSVGSvgView>
      </View>
    </View>
  </View>,
  <Modal
    animationType="none"
    hardwareAccelerated={false}
    onRequestClose={[Function]}
    transparent={true}
    visible={true}
  >
    <View
      accessibilityState={
        {
          "busy": undefined,
          "checked": undefined,
          "disabled": undefined,
          "expanded": undefined,
          "selected": undefined,
        }
      }
      accessibilityValue={
        {
          "max": undefined,
          "min": undefined,
          "now": undefined,
          "text": undefined,
        }
      }
      accessible={true}
      collapsable={false}
      focusable={false}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        {
          "flex": 1,
          "opacity": 1,
        }
      }
    >
      <View
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessible={true}
        focusable={false}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          [
            {
              "alignSelf": "center",
            },
            {
              "backgroundColor": "transparent",
              "paddingHorizontal": 12,
              "position": "absolute",
              "width": "100%",
            },
            {
              "top": 45,
            },
          ]
        }
      >
        <View
          style={
            {
              "backgroundColor": "#151823",
              "borderRadius": 8,
              "paddingHorizontal": 12,
            }
          }
        >
          <View
            accessibilityLabel="All rooms"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "opacity": 1,
                },
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "borderBottomColor": "#3E404B",
                    "padding": 7.2,
                    "paddingVertical": 16,
                  },
                  {
                    "borderBottomWidth": 1,
                  },
                ],
              ]
            }
            testID="All rooms"
          >
            <View>
              <Text
                style={
                  {
                    "color": "#ffffff",
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "includeFontPadding": false,
                    "lineHeight": 26,
                  }
                }
              >
                All rooms
              </Text>
              <Text
                style={
                  {
                    "color": "#707785",
                  }
                }
              >
                Show all rooms
              </Text>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "opacity": 1,
                  },
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#26D2E8",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "height": 20,
                      "justifyContent": "center",
                      "marginRight": 8,
                      "width": 20,
                    },
                    false,
                    undefined,
                  ]
                }
                testID="checkbox_checked"
              />
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "flex": 1,
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 14,
                      "includeFontPadding": false,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
          <View
            accessibilityLabel="Group Chats"
            accessibilityRole="button"
            accessibilityState={
              {
                "busy": undefined,
                "checked": undefined,
                "disabled": undefined,
                "expanded": undefined,
                "selected": undefined,
              }
            }
            accessibilityValue={
              {
                "max": undefined,
                "min": undefined,
                "now": undefined,
                "text": undefined,
              }
            }
            accessible={true}
            collapsable={false}
            focusable={true}
            onBlur={[Function]}
            onClick={[Function]}
            onFocus={[Function]}
            onResponderGrant={[Function]}
            onResponderMove={[Function]}
            onResponderRelease={[Function]}
            onResponderTerminate={[Function]}
            onResponderTerminationRequest={[Function]}
            onStartShouldSetResponder={[Function]}
            style={
              [
                {
                  "opacity": 1,
                },
                [
                  {
                    "flexDirection": "row",
                  },
                  {
                    "justifyContent": "space-between",
                  },
                  {
                    "borderBottomColor": "#3E404B",
                    "padding": 7.2,
                    "paddingVertical": 16,
                  },
                  false,
                ],
              ]
            }
            testID="Group Chats"
          >
            <View>
              <Text
                style={
                  {
                    "color": "#ffffff",
                    "fontFamily": "OpenSans-SemiBold",
                    "fontSize": 16,
                    "includeFontPadding": false,
                    "lineHeight": 26,
                  }
                }
              >
                Group Chats
              </Text>
              <Text
                style={
                  {
                    "color": "#707785",
                  }
                }
              >
                Show your group chats
              </Text>
            </View>
            <View
              accessibilityRole="button"
              accessibilityState={
                {
                  "busy": undefined,
                  "checked": undefined,
                  "disabled": undefined,
                  "expanded": undefined,
                  "selected": undefined,
                }
              }
              accessibilityValue={
                {
                  "max": undefined,
                  "min": undefined,
                  "now": undefined,
                  "text": undefined,
                }
              }
              accessible={true}
              collapsable={false}
              focusable={true}
              onBlur={[Function]}
              onClick={[Function]}
              onFocus={[Function]}
              onResponderGrant={[Function]}
              onResponderMove={[Function]}
              onResponderRelease={[Function]}
              onResponderTerminate={[Function]}
              onResponderTerminationRequest={[Function]}
              onStartShouldSetResponder={[Function]}
              style={
                [
                  {
                    "opacity": 1,
                  },
                  [
                    {
                      "alignItems": "center",
                      "flexDirection": "row",
                    },
                    undefined,
                  ],
                ]
              }
            >
              <View
                style={
                  [
                    {
                      "alignItems": "center",
                      "borderColor": "#26D2E8",
                      "borderRadius": 4,
                      "borderWidth": 1,
                      "height": 20,
                      "justifyContent": "center",
                      "marginRight": 8,
                      "width": 20,
                    },
                    {
                      "backgroundColor": "#26D2E8",
                    },
                    undefined,
                  ]
                }
                testID="checkbox_checked"
              >
                <RNSVGSvgView
                  align="xMidYMid"
                  bbHeight="100%"
                  bbWidth="100%"
                  color="#141925"
                  fill="#141925"
                  focusable={false}
                  height="100%"
                  meetOrSlice={0}
                  minX={0}
                  minY={0}
                  style={
                    [
                      {
                        "backgroundColor": "transparent",
                        "borderWidth": 0,
                      },
                      {},
                      {
                        "flex": 0,
                        "height": "100%",
                        "width": "100%",
                      },
                    ]
                  }
                  testID="svg-icon-checkFat"
                  tintColor="#141925"
                  vbHeight={24}
                  vbWidth={24}
                  width="100%"
                  xml="<svg fill="currentColor" height="24" viewBox="0 0 24 24" width="24" xmlns="http://www.w3.org/2000/svg"><path d="m21.796 7.54594-12.00004 11.99996c-.10452.1049-.22871.1881-.36546.2449-.13674.0568-.28335.086-.43142.086-.14806 0-.29467-.0292-.43142-.086-.13674-.0568-.26094-.14-.36545-.2449l-5.25-5.25c-.10465-.1046-.18766-.2288-.24429-.3656-.05664-.1367-.08579-.2832-.08579-.4312s.02915-.2946.08579-.4313c.05663-.1367.13964-.261.24429-.3656.10464-.1047.22888-.1877.36561-.2443s.28327-.0858.43126-.0858c.148 0 .29454.0292.43127.0858s.26096.1396.36561.2443l4.45406 4.454 11.20408-11.20214c.2113-.21134.498-.33007.7969-.33007.2988 0 .5855.11873.7968.33007.2114.21135.3301.49799.3301.79688 0 .29888-.1187.58553-.3301.79687z" /></svg>
"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <RNSVGGroup
                    fill={
                      {
                        "payload": 4279507237,
                        "type": 0,
                      }
                    }
                    propList={
                      [
                        "fill",
                      ]
                    }
                  >
                    <RNSVGPath
                      d="m21.796 7.54594-12.00004 11.99996c-.10452.1049-.22871.1881-.36546.2449-.13674.0568-.28335.086-.43142.086-.14806 0-.29467-.0292-.43142-.086-.13674-.0568-.26094-.14-.36545-.2449l-5.25-5.25c-.10465-.1046-.18766-.2288-.24429-.3656-.05664-.1367-.08579-.2832-.08579-.4312s.02915-.2946.08579-.4313c.05663-.1367.13964-.261.24429-.3656.10464-.1047.22888-.1877.36561-.2443s.28327-.0858.43126-.0858c.148 0 .29454.0292.43127.0858s.26096.1396.36561.2443l4.45406 4.454 11.20408-11.20214c.2113-.21134.498-.33007.7969-.33007.2988 0 .5855.11873.7968.33007.2114.21135.3301.49799.3301.79688 0 .29888-.1187.58553-.3301.79687z"
                      fill={
                        {
                          "payload": 4278190080,
                          "type": 0,
                        }
                      }
                    />
                  </RNSVGGroup>
                </RNSVGSvgView>
              </View>
              <Text
                style={
                  [
                    {
                      "color": "#ffffff",
                      "flex": 1,
                      "fontFamily": "OpenSans-Regular",
                      "fontSize": 14,
                      "includeFontPadding": false,
                    },
                    undefined,
                  ]
                }
              />
            </View>
          </View>
        </View>
      </View>
    </View>
  </Modal>,
]
`;
